packages/components/addon/components/hds/table/index.hbs

```diff
@@ -20,7 +24,7 @@
   </thead>
   <tbody class="hds-table__tbody" data-test-table-body>
     {{#if this.isSortableTable}}
-      {{#each (sort-by this.getSortCriteria @model) as |data|}}
+      {{#each (sort-by this._getSortCriteria @model) as |data|}}
```

Comment:
hmmm. maybe let's discuss, since this is a common way to indicate private API and a signal that developers should not use this and cannot rely on it (in Ember, private API can change without an RFC).

Reply:
There's plenty of similar methods/getters in the other components, that are not directly exposed, and this didn't come up before, so I would avoid to introduce a new pattern, unless we change it across all the other components, which is not feasible for me.
Plus, when you say _signal that developers should not use this and cannot rely on it_ what do you mean? Us or the consumers? If it's us, I don't see the problem; if it's the consumers, I don't understand how they could invoke that function from outside, unless they do some convoluted thing like this:
```
<Hds::Table as |T|>
  <some component that calls T._getSortCriteria />
```

Reply:
I will switch it back for now, but I strongly disagree with the premise that just because we hadn't discussed it until now, it should not be allowed. There are a lot of things we haven't discussed yet because we haven't needed to. We should be able to iterate and improve over time.

Reply:
For clarity: I am not saying it's not allowed. I am saying that **in the context of this PR** doesn't make sense because:
- it introduces a new pattern in the codebase, only for this component, without tackling all the other components
- it's not clear enough (at least for me, from the description above) what are the reasons why this would be beneficial (and for who)

Maybe open a new issue in GitHub, in which you explain why we should change the way in which we declare the getters/setters in the codebase, differentiating between public and private methods (and what that means in the context of a UI component) and what are the benefits of changing the current approach (what are the downsides of the current? what the upside of the proposed one?)